pkg/export/debug/debug.go

```diff
@@ -10,18 +11,90 @@ import (
 	"github.com/grafana/beyla/pkg/internal/request"
 )
 
+// XXX deprecated (REMOVE) - use TracePrinter instead
 type PrintEnabled bool
 
 func (p PrintEnabled) Enabled() bool {
 	return bool(p)
 }
 
-func PrinterNode(e PrintEnabled) pipe.FinalProvider[[]request.Span] {
+type TracePrinter uint
+
+const (
```

Comment:
Maybe think of a way how we can only have one set of values for the constants. We now have this duplicity, an enum of numbers and string versions of the same thing.

Reply:
The other way would be relying solely on string constants, but that comes with its own set of trade-offs (for instance, more overhead in relation to numerical types), unless Go reflection can help us here?

Reply:
I don't think we need to use reflection. Go is great at functional programming. You can have an initial switch on the string constants which returns print provider function, it's done once on setup and from than on, there are no branches to provide the printing.

Reply:
So, there's a little bit more to that. It helps if the type `TracePrinter` has a few attached methods, namely:
 - TracePrinter.Enabled()
 - TracePrinter.Valid()

because it makes code in `Validate()` trivial to write and comprehend, for example:

```
>   if c.Printer.Enabled() && c.TracePrinter.Enabled() {                                                                                                                                                                 
>   >   return ConfigError("print_traces and trace_printer are mutually exclusive, use trace_printer instead")
>   }
```

Otherwise I think we'd need to manually evaluate a string value inside `config.go` which IMHO is much worse as we now have logic scattered in two places (and a string based API that offers no strong type guarantees).

The way I see it, if we want to remove the mapping from `int` to `string`, is to simply make the underlying type of `TracePrinter` be `string`. You make a nice point this is just code that is run once at startup, and the only other thing we lose is the automatic error checking when deserializing the option, as in:

```
time=2024-07-22T11:15:27.564-04:00 level=ERROR msg="wrong configuration" !BADKEY="reading env vars: env: parse error on field \"TracePrinter\" of type \"debug.TracePrinter\": unknown TracePrinter value \"json2\""
```

but that is covered in `Validate()` - in a nutshell, trade-offs.